### PR TITLE
update e2e test script and setup_all.sh

### PR DESCRIPTION
**Pull request by jonchenn**
- Fixed setup_all script 
- Update e2e test without creating new project.
- Add docs to set up template e2e test. (this will be used by GHA later)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -97,6 +97,7 @@ Run the following to create a new Google Cloud project, or you can log in to Goo
 export PROJECT_ID=<my-project-id>
 gcloud projects create $PROJECT_ID
 gcloud config set project $PROJECT_ID
+gcloud auth application-default set-quota-project $PROJECT_ID
```
Comment (lavjain) [inline README.md:101]: Is `gcloud auth application-default login` not required?

File: docs/TROUBLESHOOTING.md [hunk 2]
```diff
@@ -24,4 +24,40 @@
...
+
+    # This will show the following:
+    {
+      "client_id": "<hash-id>.apps.googleusercontent.com",
+      "client_secret": "<client_secret>",
+      "quota_project_id": "solutions-template-e2etest",
+      "refresh_token": "<refresh_token-hash>",
+      "type": "authorized_user"
+    }
+    ```
```
Comment (lavjain) [inline docs/TROUBLESHOOTING.md:58]: Also check environment variable APPLICATION_DEFAULT_CREDENTIALS

File: e2e/template_e2e_tests/run_e2e_test.sh [hunk 3]
```diff
@@ -66,29 +80,31 @@ echo yes | sh ./build_tools/build_template.sh
...
 export OUTPUT_FOLDER=".test_output"
-export PROJECT_ID=solutemp-e2e-$(uuidgen | head -c 8 | awk '{print tolower($0)}')
 export ADMIN_EMAIL=$(gcloud auth list --filter=status:ACTIVE --format='value(account)')
 pip3 install pytest --no-input
-echo "PROJECT_ID=$PROJECT_ID"
 
-### Create a new Google Cloud project:
+# Create a new Google Cloud project
 create_new_project() {
+  export PROJECT_ID=solutemp-e2e-$(uuidgen | head -c 8 | awk '{print tolower($0)}')
```
Comment (lavjain) [inline e2e/template_e2e_tests/run_e2e_test.sh:88]: For creating a new project, the user should be expected to unset PROJECT_ID in the environment (as an extra step)

File: README.md (hunk 1 above)
Comment (jonchenn) [reply README.md:101]: Good catch, added.

File: docs/TROUBLESHOOTING.md (hunk 2 above)
Comment (jonchenn) [reply docs/TROUBLESHOOTING.md:58]: I assume you mean GOOGLE_APPLICATION_CREDENTIALS. Added to the doc.

File: e2e/template_e2e_tests/run_e2e_test.sh (hunk 3 above)
Comment (jonchenn) [reply e2e/template_e2e_tests/run_e2e_test.sh:88]: That won't be necessary, as the script takes an argument to override the $PROJECT_ID. It doesn't rely on if there's any pre-existing PROJECT_ID.

**Review by lavjain — APPROVED** [review]

**Pull request merged**